### PR TITLE
Add transfer hotspot transaction

### PR DESCRIPTION
**Pull request by mrallen1**
Problem to solve: enable hotspot transfers between owners

Clubhouse: https://app.clubhouse.io/hlm/story/8742/transfer-ownership-transaction

### CODE REVIEW

File: src/blockchain_txn_transfer_hotspot_v1.proto [hunk 1]
```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+package helium;
+
+message blockchain_txn_transfer_hotspot_v1 {
+    bytes gateway = 1;
+    bytes current_owner = 2;
```
Comment (Vagabond) [inline src/blockchain_txn_transfer_hotspot_v1.proto:6]: The ledger contains the current owner so this is likely redundant.

File: src/blockchain_txn_transfer_hotspot_v1.proto [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    bytes payer = 3;
+    bytes new_owner = 4;
+    bytes current_owner_signature = 5;
+    bytes new_owner_signature = 6;
+    bytes payer_signature = 7;
+    uint64 hnt_to_transfer = 8;
+    uint64 fee = 9;
+    uint64 staking_fee = 10;
```
Comment (Vagabond) [inline src/blockchain_txn_transfer_hotspot_v1.proto:14]: I don't think there's a staking fee for this?

File: src/blockchain_txn_transfer_hotspot_v1.proto [hunk 3]
```diff
@@ -0,0 +1,12 @@
+syntax = "proto3";
+package helium;
+
+message blockchain_txn_transfer_hotspot_v1 {
+    bytes gateway = 1;
+    bytes seller = 2;
+    bytes buyer = 3;
+    bytes seller_signature = 4;
+    bytes buyer_signature = 5;
+    uint64 hnt_to_seller = 6;
```
Comment (abhay) [inline src/blockchain_txn_transfer_hotspot_v1.proto:10]:
```suggestion
    uint64 amount_to_seller = 6;
```

Should this be an `amount` to match the others? And also in bones and not hnt?

**Comment by vihu** [comment]
Missing imports and oneof definition in `blockchain_txn.proto`

**Comment by vihu** [comment]
> Missing imports and oneof definition in `blockchain_txn.proto`

Added those

File: src/blockchain_txn_transfer_hotspot_v1.proto (hunk 3 above)
Comment (mrallen1) [reply src/blockchain_txn_transfer_hotspot_v1.proto:10]: @evanmcc Do you have thoughts about this?

File: src/blockchain_txn.proto [hunk 4]
```diff
@@ -56,6 +57,7 @@ message blockchain_txn {
         blockchain_txn_payment_v2 payment_v2 = 24;
         blockchain_txn_price_oracle_v1 price_oracle_submission = 25;
         blockchain_txn_gen_price_oracle_v1 gen_price_oracle = 26;
+        blockchain_txn_transfer_hotspot_v1 transfer_hotspot = 27;
```
Comment (abhay) [inline src/blockchain_txn.proto:60]:
```suggestion
        blockchain_txn_transfer_hotspot_v1 transfer_hotspot_v1 = 27;
```
?

Comment (mrallen1) [reply src/blockchain_txn.proto:60]: We usually don't version `v1`, but `v2` or later gets one. At least that's been the pattern so far.

Comment (abhay) [reply src/blockchain_txn.proto:60]: Makes sense!

**Pull request merged**